src/simpleDetFunc.c

```diff
@@ -224,7 +224,23 @@ void rollSigma(int startIndex, int interval, float scaleFactor,
 
 
 
+#define EXP_A 184
+#define EXP_C 16249 
 
+float EXP_APPROX(float x)
+{
+	//union allows us to treat the same 4 bytes of memory as both a float and 2 shorts
+	union {
+		float f;
+		struct {
+			short j, i; //if on big-endian architecture, j, i must be flipped to i, j
```

Comment:
If we make the additional modification I proposed in my comment for the CMakefile above, then a macro  called `BIG_ENDIAN` will be defined big-endian machines. Then we can modify the struct declaration here as:

```
struct{
#ifdef BIG_ENDIAN
        short i, j;
#else
        short j, i;
#endif
} s;
```

If we wanted to be more careful, we could include `<inttypes.h>` (which became part of the standard library in C99) and use the `int16_t` type in place of `short`.